add DuckDB::LogicalType#type

otegami opened the pull request:
refs: GH-690

In this PR, we add DuckDB::LogicalType#type, which returns logical type's type.
- [duckdb_type duckdb_get_type_id(duckdb_logical_type type);](https://duckdb.org/docs/api/c/api.html#duckdb_get_type_id)

This is one of the steps for supporting duckdb_logical_column_type C API.

otegami commented on test/duckdb_test/column_test.rb:
```diff
@@ -145,6 +145,11 @@ def test_logical_type
       assert(logical_types.all? { |logical_type| logical_type.is_a?(DuckDB::LogicalType) })
     end
 
+    def test_logical_type_type
```
I'm thinking about where we should test this method because test data will be duplicated with `test/duckdb_test/column_test.rb`. Should we define it in `test/duckdb_test/logical_type_test.rb`?

suketa replied:
@otegami 

This test is for DuckDB::LogicalType#type, right?
Then, I think it should be in test/duckdb_test/logical_type_test.rb.

otegami replied:
Thank you so much. I think it should be there after rethinking it now too.

suketa reviewed (commented):
@otegami 
Thank you for your progress and sorry for being late to review.

otegami commented:
@suketa 
Please don't worry about it at all. I'm very grateful just for your review.
Although my progress is slow, I hope to contribute little by little.

---

I've just rebased and cleaned up my commits.

suketa reviewed (approved):
Thank you
